### ISSUE TITLE
make_mth5 : Factoring and expanding use cases

### ISSUE THREAD

**Issue by kkappler** [kujaku11/mth5#76]
I started testing the methods in MakeMTH5 in the make_mth5.py  module on CAS04, which relates to aurora issue#31.  
I ran into three issues:
1. I intermittently encounter this error and am not able to reproduce it:
make_mth5 Value Error: The current client does not have a dataselect service.
2. Make_mth5 does not support partial files:
Say there are four runs, but I only want to pull the data for one of them, the method fails because there is a different number of runs in the metadata than in the this fails.
3. Failed for 0.2.0 version of mth5.

**Comment by kkappler** [comment]
@timronan : I pushed a branch called issue_76_make_mth5_factoring with a copy of make_mth5.py called mth5/clients/make_mth5_rev_002.py with some suggested changes, it also depends on helper_functions.py.  You can meld this against make_mth5.py to see what I changed.

FWIW I had to suppress a flake compaint:
C901 'MakeMTH5.make_mth5_from_fdsnclient' is too complex (25)

We can take a deeper look on Wednesday hopefully.
Let's try to re

**Comment by kkappler** [comment]
The modifications I made to make_mth5 were on the right track, but they still left an element. If I provided a csv with two time series from the same channel (and station), having a different start times, I wound up with two instances of the station object in the internal bookkeeping of make_mth5. It maybe that when everything gets merged at the end this doesn't create issues, but there is something unsatisfying about it.

If the user provides a data frame specifying SNCLs (with start and endtimes) for each time series, the program should be able to identify how many unique Networks and how many unique Stations are involved, and then iterate over unique networks and over unique stations.

This shouldn't be too hard to implement, although I haven't done it.  But because the obspy network ( & station) objects  have an __eq_(other) method, we can actually detect unique networks (& stations) while accumulating the lists. The trick is how to manage this information. How about adding two new columns to the data_request_dataframe that are generated on the fly? One column called for example network_uuid and one called station_uuid. These get populated during make_network_inventory and make_station_inventory.

So instead of:

network_inventory = make_network_inventory(df, client)
station_inventory = make_station_inventory(df, client)
we do:

network_inventory, df = make_network_inventory(df, client)
station_inventory, df = make_station_inventory(df, client)
i.e. the df comes back with the new columns populated.  Then when the big loop over channels happens it can loop over network_uuid, station_uuid, channel, starttime, rather than network, station, channel, starttime and the code logic should stay pretty easy to keep clean.
